### PR TITLE
Add changes for v1.2.3 release

### PR DESCRIPTION
**Pull request by tmcgilchrist**

### CODE REVIEW

**Comment by tmcgilchrist** [comment]
@talex5 you'll need to do the dune-release I don't have permissions for it.

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+### v1.2.3
+
+- Update to cmdliner 1.1.0 (@MisterDA #249)
+
+- Fix tests on Windows, avoid using Unix domain sockets (@talex5 #251)
+
+- Fix warning on Unit.ENOTCONN under macos (@talex5 #252)
```
Comment (talex5) [inline CHANGES.md:7]:
```suggestion
- Fix warning on `Unix.ENOTCONN` under macos (@talex5 #252)
```
BTW, are you generating the changes updates manually? I have a little script to do this that you might like...

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+
+- Add Capnp_rpc_unix.Cap_file.save_uri (@talex5 #253)
```
Comment (talex5) [inline CHANGES.md:9]:
```suggestion
- Add `Capnp_rpc_unix.Cap_file.save_uri` (@talex5 #253)
```

**Review by talex5 — COMMENTED** [review]
My script also generated these:
```
- Update for opam 2.1+ installations (@frumioj #259).

- Install Cap'n Proto binaries in Windows GHA (@MisterDA #248).
```

File: CHANGES.md (hunk 1 above)
Comment (tmcgilchrist) [reply CHANGES.md:7]: Yes, I'm doing this manually by browsing the diff between tags and master. 
Automation would be appreciated.

**Comment by tmcgilchrist** [comment]
> - Update for opam 2.1+ installations (@frumioj #259).
> - Install Cap'n Proto binaries in Windows GHA (@MisterDA #248).

I didn't feel either change warranted a changelog mention, they're useful CI and docs updates but don't effect the library? WDYT

**Comment by talex5** [comment]
Thanks! They're not essential in the changelog, but it's nice to credit people for their contributions.

**Pull request merged**